Sqlite: add a test of cascading delete

rvl opened the pull request:
Relates to #154, base branch PR is #247.

# Overview

This adds "cascading" deletes before every Sqlite test. It also demonstrates a good way of speeding up tests -- by performing DB migrations only once per test suite.

Once the other DBLayer methods are implemented, we will be able to see whether cascading deletes actually work under Sqlite.

jonathanknowles commented on cardano-wallet.cabal:
```diff
@@ -556,8 +552,8 @@ test-suite integration
       NoMonomorphismRestriction
-      OverloadedStrings
       OverloadedLabels
+      OverloadedStrings
```
This line movement was a side-effect of running "sort -u".

rvl committed:
Sqlite: add a test of cascading delete

This will show whether cascading deletes work under sqlite -- once the
functions to write other tables are implemented.

paweljakubas commented on lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs:
```diff
@@ -34,29 +36,29 @@ import Data.ByteString
 
 spec :: Spec
-spec = do
+spec = beforeAll (newDBLayer Nothing) $ beforeWith cleanDB $ do
```
good approach, instantiate db just ONCE before first spec item, and then clean db before EVERY spec item
